src/adapter/src/catalog/transact.rs

```diff
@@ -206,10 +206,9 @@ impl Catalog {
             .cloned()
             .map(ObjectId::Item)
             .collect();
-        self.object_dependents(&temp_ids, conn_id)
-            .into_iter()
-            .map(Op::DropObject)
-            .collect()
+        let drop_ids = self.object_dependents(&temp_ids, conn_id);
+
+        vec![Op::DropObjects(drop_ids)]
```

Comment:
I think you can probably just update the return type to `Op` (no `Vec`).

Reply:
Good call, had to do a bit more refactoring, but updated!